### PR TITLE
[bugfix] Check that oldData is defined before trying to migrate

### PR DESCRIPTION
**Pull request by salazarm**
## Summary

FIxes issue where launchpad config gets wiped out whenever you load the launchpad.

## Test Plan

 
Made a change to launchpad config. Reload the page. See that the launchpad config sticks around.

## Checklist

- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Comment by gibsondan** [comment]
This makes sense to me - I assume the summary/test plan is placeholder, given that we'll likely hotfix this would love to see a more fleshed out test plan that verifies that the issue no longer occurs before accepting

**Review by salazarm — COMMENTED** [review]
Updated. Was having technical difficulties for a second

**Review by hellendag — APPROVED** [review]

**Comment by salazarm** [comment]
**Graphite Merge Job**

Current status: 🔄 Waiting for checks to pass

This pull request is currently waiting for CI and mergeability checks to pass before being merged.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/7537?utm_source=stack-comment).


Job Reference: f2Xnqp5h5pmjvCufe9cW

File: js_modules/dagit/packages/core/src/app/__tests__/ExecutionSessionStorage.test.tsx
Comment (hellendag) [inline js_modules/dagit/packages/core/src/app/__tests__/ExecutionSessionStorage.test.tsx:30]: Looks like you can probably use the real `window.localStorage` instead of a mock after all, since we don't need to spy on it or do other mock stuff with it.

File: js_modules/dagit/packages/core/src/app/__tests__/ExecutionSessionStorage.test.tsx
Comment (hellendag) [inline js_modules/dagit/packages/core/src/app/__tests__/ExecutionSessionStorage.test.tsx:40]: Can use `RepoAddress` as a type instead of `typeof REPO_ADDRESS`.

File: js_modules/dagit/packages/core/src/app/__tests__/ExecutionSessionStorage.test.tsx
Comment (hellendag) [inline js_modules/dagit/packages/core/src/app/__tests__/ExecutionSessionStorage.test.tsx:38]: Prefer "job" nomenclature

File: js_modules/dagit/packages/core/src/app/__tests__/ExecutionSessionStorage.test.tsx
Comment (hellendag) [inline js_modules/dagit/packages/core/src/app/__tests__/ExecutionSessionStorage.test.tsx:73]: Maybe a little odd to `expect` within the render path.

File: js_modules/dagit/packages/core/src/app/__tests__/ExecutionSessionStorage.test.tsx
Comment (hellendag) [inline js_modules/dagit/packages/core/src/app/__tests__/ExecutionSessionStorage.test.tsx:71]: Since `newFormat` is already evaluated and we never access `BASE_PATH` again, is this necessary?

File: js_modules/dagit/packages/core/src/app/ExecutionSessionStorage.tsx
Comment (hellendag) [inline js_modules/dagit/packages/core/src/app/ExecutionSessionStorage.tsx:183]: It occurs to me that we should maybe be doing side-effectful stuff like this in an effect instead.

**Review by hellendag — COMMENTED** [review]
Non-blocking comments.

File: js_modules/dagit/packages/core/src/app/__tests__/ExecutionSessionStorage.test.tsx
Comment (salazarm) [reply js_modules/dagit/packages/core/src/app/__tests__/ExecutionSessionStorage.test.tsx:71]: oh yeah true

File: js_modules/dagit/packages/core/src/app/ExecutionSessionStorage.tsx
Comment (salazarm) [reply js_modules/dagit/packages/core/src/app/ExecutionSessionStorage.tsx:183]: Yeah we can. The only downside is it'll be an extra render until the data shows up

Comment (salazarm) [reply js_modules/dagit/packages/core/src/app/ExecutionSessionStorage.tsx:183]: https://github.com/dagster-io/dagster/pull/7540/files

**Pull request merged**
